### DISCUSSION TITLE
Why does a timeout occur when publishing a stream via RTSP?

### DISCUSSION THREAD

**Discussion by rbnrtg** [bluenviron/mediamtx#3247] [Questions]
### Question

Hi,

When I publish an RTSP stream on the server, if some time passes without reading the published stream, the connection that publishes the video stream is closed, as shown in the server logs:

2024/04/11 08:35:56 INF [RTSP] [session c9c61d8f] created by 172.18.0.5:51916
2024/04/11 08:35:57 INF [RTSP] [session c9c61d8f] is publishing to path 'video', 1 track (H264)
2024/04/11 08:35:58 INF [WebRTC] [session 0102690b] created by 192.168.0.106:39838
2024/04/11 08:35:58 INF [WebRTC] [session 0102690b] peer connection established, local candidate: host/udp/192.168.0.233/8189, remote candidate: prflx/udp/192.168.0.106/53217
2024/04/11 08:35:58 INF [WebRTC] [session 0102690b] is reading from path 'video', 1 track (H264)
2024/04/11 08:36:20 INF [WebRTC] [session 0102690b] closed: peer connection closed
2024/04/12 05:57:11 INF [RTSP] [conn 172.18.0.5:51916] closed: read tcp 172.18.0.2:8554->172.18.0.5:51916: i/o timeout
2024/04/12 05:57:11 INF [RTSP] [session c9c61d8f] destroyed: not in use

The pipeline I am publishing is as follows:

gst-launch-1.0 pylonsrc ! video/x-raw,width=1920 ,height=1080 ,format=RGB,framerate=30/1 ! videoconvert ! queue ! x264enc bitrate=26000 speed-preset=ultrafast tune=0 ! video/x-h264,profile=main ! rtspclientsink protocols=tcp timeout=0 tcp-timeout=0 latency=0 location=rtsp://127.0.0.1:8554/video

How to avoid this connection closure and keep the stream published all the time?

Than you.

**Comment by aler9** [comment]
Hello, the fact that there are no WebRTC readers is unrelated from the fact that the RTSP session of the publisher gets closed.

The RTSP session gets closed because its associated RTSP connection is closed due to a i/o timeout:

```
2024/04/12 05:57:11 INF [RTSP] [conn 172.18.0.5:51916] closed: read tcp 172.18.0.2:8554->172.18.0.5:51916: i/o timeout
```

If there's a i/o timeout, it means that GStreamer / pylonsrc is not sending packets anymore. Check why this happens by looking at GStreamer logs.

[a comment by an automated account is not shown]
